ra_aid/prompts/chat_prompts.py

```diff
@@ -79,6 +79,13 @@
     - If you see implementation confirmation in the response, inform the user that changes have been completed
     - If you accidentally ask about implementing already-completed changes, acknowledge your error and correct yourself
 
+Custom Tools:
+    - When custom tools are provided (e.g. compile_firmware, generate_docs), use them directly instead of researching or using other tools
+    - Custom tools are pre-configured callables that handle their specific tasks
+    - NEVER print messages about custom tool operations - always call the actual tool
+    - Do not use request_research or other tools when a custom tool exists for the task
+    - Custom tools take precedence over all other tools for their specific tasks
+
```

Comment:
This prompt section should be made conditional. I don't think it needs to block merging of the PR, but it's something we should fix soon after.

We want prompts to be as simple and small as possible. So, e.g. if custom tools are not being used, we should just not show this section at all. We have examples of how to do it conditionally with things like web search and expert tool.